### PR TITLE
chore: revive max blob size decorator

### PR DESCRIPTION
**Pull request by rootulp**
Closes https://github.com/celestiaorg/celestia-app/issues/3468 by reviving the max blob size decorator and making both decorators version aware so that:
- v1 uses max blob size decorator
- v2 uses max blob share decorator

### CODE REVIEW

**Review by rach-id — APPROVED** [review]
LGTM utAck

left some optional nits

File: x/blob/ante/max_total_blob_size_ante.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+	govParam := d.k.GovMaxSquareSize(ctx)
+	return min(upperBound, int(govParam))
+}
+
+// getTotal returns the sum of the given sizes.
+func getTotal(sizes []uint32) (sum int) {
+	for _, size := range sizes {
+		sum += int(size)
+	}
+	return sum
```
[the remark this answers is not in the document]
Comment (ninabarbakadze) [reply x/blob/ante/max_total_blob_size_ante.go:82]: I think the original version is more readable

**Review by ninabarbakadze — APPROVED** [review]
🚢

**Review by evan-forbes — APPROVED** [review]
lgtm and makes sense.

In this specific case, meaning when the antehandler is only ran during checktx, I think after all network are upgraded we will actually able to get rid of this one.

File: x/blob/ante/max_total_blob_size_ante.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (rootulp) [reply x/blob/ante/max_total_blob_size_ante.go:82]: +1 we should not use naked returns

Ref: https://go.dev/tour/basics/7

File: x/blob/ante/max_total_blob_size_ante.go [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// MaxTotalBlobSizeDecorator helps to prevent a PFB from being included in a
+// block but not fitting in a data square.
+type MaxTotalBlobSizeDecorator struct {
+	k BlobKeeper
+}
+
+func NewMaxBlobSizeDecorator(k BlobKeeper) MaxTotalBlobSizeDecorator {
```
[the remark this answers is not in the document]
Comment (rootulp) [reply x/blob/ante/max_total_blob_size_ante.go]: 594c2c2e30dca2bb9e17a9e0342509d22971798c

File: x/blob/ante/max_total_blob_size_ante.go [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+func (d MaxTotalBlobSizeDecorator) getMaxSquareSize(ctx sdk.Context) int {
+	// TODO: fix hack that forces the max square size for the first height to
+	// 64. This is due to our fork of the sdk not initializing state before
+	// BeginBlock of the first block. This is remedied in versions of the sdk
+	// and comet that have full support of PreparePropsoal, although
+	// celestia-app does not currently use those. see this PR for more details
+	// https://github.com/cosmos/cosmos-sdk/pull/14505
+	if ctx.BlockHeader().Height <= 1 {
+		return int(appconsts.DefaultGovMaxSquareSize)
+	}
```
[the remark this answers is not in the document]
Comment (rootulp) [reply x/blob/ante/max_total_blob_size_ante.go:70]: https://github.com/celestiaorg/celestia-app/issues/3495

**Review by rach-id — APPROVED** [review]

**Review by evan-forbes — APPROVED** [review]

**Pull request merged**